packages/story-editor/src/components/videoTrim/useVideoTrimMode.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useFeature } from 'flagged';
+import { useCallback, useMemo } from '@web-stories-wp/react';
+
+/**
+ * Internal dependencies
+ */
+import { useCanvas, useStory } from '../../app';
+
+function useVideoTrimMode() {
+  const isVideoTrimEnabled = useFeature('enableVideoTrim');
+  const { isEditing, isTrimming, setEditingElementWithState, clearEditing } =
+    useCanvas(
+      ({
+        state: { isEditing, editingElementState: { isTrimming } = {} },
+        actions: { setEditingElementWithState, clearEditing },
+      }) => ({
+        isEditing,
+        isTrimming,
+        setEditingElementWithState,
+        clearEditing,
+      })
+    );
+  const { selectedElement } = useStory(({ state: { selectedElements } }) => ({
+    selectedElement: selectedElements.length === 1 ? selectedElements[0] : null,
+  }));
+
+  const toggleTrimMode = useCallback(() => {
+    if (isEditing) {
+      clearEditing();
+    } else {
+      setEditingElementWithState(selectedElement.id, {
+        isTrimming: true,
+      });
+    }
+  }, [isEditing, clearEditing, setEditingElementWithState, selectedElement]);
+
+  const hasTrimMode = useMemo(
+    () => selectedElement?.type === 'video' && isVideoTrimEnabled,
```

Comment:
We need to make sure that user do not trim video if the video is current
- Transcoding
- Muting 
- Not local. 

https://github.com/google/web-stories-wp/pull/8972/files#diff-d581131a68dd353c626d1714ea3b4d65115979020ed22b61c7e5d4a0a8fed0ddR116-R123

Reply:
Yeah, that makes sense. A non-local video can never be trimmed (of course) and a transcoding or muting video can only be trimmed once the current process is done (which will turn those flags false). I'll add that, thanks!

Reply:
I added in my PR #8972 . When you implement this, do feel free to remove my logic and load it from `hasTrimMode`

Reply:
Implemented in a8459cbc7d - would that do it? I also added that you couldn't trim a video while it's already trimming, which seems trivial as well.

Reply:
I missed it off my list but I implemented it 

```
 return (
      isSingleElement &&
      isVideoTrimEnabled &&
      !local &&
      !isMuting &&
      !isTranscoding &&
      !isTrimming
    );
```

Reply:
Fixed in 8162d8f69a.